controller/utils.py

```diff
@@ -257,3 +261,13 @@ def sanitize_prometheus_metric_label_name(val):
     val = re.sub(first_letter, "_", val, count=1)
     val = re.sub(all_letters, "_", val)
     return val
+
+
+def k8s_timestamp_to_utc_datetime(timestamp: str) -> datetime:
+    return datetime.fromisoformat(
```

Comment:
Did you consider using https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse ? It's what https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat recommends for parsing arbitrary iso8601 strings. 

Not sure what the risk of getting a format that this code here doesn't cover is.

Reply:
> Did you consider using https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.isoparse

I had no clue about this... I will use this instead. I think that the timestamps in k8s will either have an offset in hours or just `Z` for UTC. So The risk of getting a code that is not covered is low - or def lower than this hacky function.